src/Engine/EmitMSIL/Replication.cs

```diff
@@ -46,10 +46,15 @@ public static IList ReplicationLogic(IEnumerable<MethodBase> mInfos, IList args,
                 //Validity.Assert(args.Count == mi.GetParameters().Length);
                 result = mi.Invoke(null, reducedArgs.ToArray());
             }
+            else if(mi.IsConstructor)
+            {
+                result = (mi as ConstructorInfo).Invoke(reducedArgs.ToArray());
+            }
             else
             {
                 result = mi.Invoke(reducedArgs[0], reducedArgs.Skip(1).ToArray());
             }
+            if (result is IList) return (IList)result;
```

Comment:
so arrays/lists should not get wrapped in yet another list?
But single values should still get wrapped ?

Reply:
Single values need to get wrapped as the return type of the method is `IList`. I think we might be able to avoid it if we can avoid calling ReplicationLogic for non-replicated cases.